com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/util/DeployedModuleCalculatorColor.java

```diff
@@ -0,0 +1,66 @@
+package com.sap.cloud.lm.sl.cf.core.cf.util;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.sap.cloud.lm.sl.cf.core.model.ApplicationColor;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+
+public class DeployedModuleCalculatorColor implements DeployedModuleCalculator {
+
+    public static final ApplicationColor COLOR_OF_APPLICATIONS_WITHOUT_SUFFIX = ApplicationColor.BLUE;
+
+    private DeployedMta deployedMta;
+
+    private ApplicationColor deployedMtaColor;
+
+    public DeployedModuleCalculatorColor(DeployedMta deployedMta, ApplicationColor deployedMtaColor) {
+        this.deployedMta = deployedMta;
+        this.deployedMtaColor = deployedMtaColor;
+    }
+
+    @Override
+    public DeployedMtaModule getLiveDeployedModule(String moduleName) {
+        if (deployedMta == null) {
+            return null;
+        }
+
+        Stream<DeployedMtaModule> deployedModules = getDeployedModules().stream()
+                                                                        .filter(deployedModule -> isModuleNameMatching(deployedModule,
+                                                                                                                       moduleName));
+
+        if (deployedMtaColor != null) {
+            return deployedModules.filter(this::hasDeployedModuleContainLiveColor)
+                                  .findFirst()
+                                  .orElse(null);
+        }
+
+        return deployedModules.findFirst()
+                              .orElse(null);
+    }
+
+    private List<DeployedMtaModule> getDeployedModules() {
```

Comment:
I don't think the amount of uses and length (as an expression) of `deployedMta.getModules` warrants the need for this method. Just use it normally and delete this